src/wp-includes/option.php

```diff
@@ -253,6 +253,88 @@ function get_option( $option, $default_value = false ) {
 	return apply_filters( "option_{$option}", maybe_unserialize( $value ), $option );
 }
 
+/**
+ * Primes specific options into the cache with a single database query.
+ *
+ * Only options that do not already exist in cache will be primed.
+ *
+ * @since 6.4.0
+ *
+ * @global wpdb $wpdb WordPress database abstraction object.
+ *
+ * @param array $options An array of option names to be primed.
+ */
+function prime_options( $options ) {
+	$alloptions     = wp_load_alloptions();
+	$cached_options = wp_cache_get_multiple( $options, 'options' );
+
+	// Filter options that are not in the cache.
+	$options_to_prime = array();
+	foreach ( $options as $option ) {
+		if ( ! isset( $cached_options[ $option ] ) && ! isset( $alloptions[ $option ] ) ) {
+			$options_to_prime[] = $option;
+		}
+	}
+
+	if ( ! empty( $options_to_prime ) ) {
+		global $wpdb;
+		$option_names = implode( "','", array_map( 'esc_sql', $options_to_prime ) );
+		$query   = "SELECT option_name, option_value FROM {$wpdb->options} WHERE option_name IN ('$option_names')";
+		$results = $wpdb->get_results( $query, OBJECT_K );
+
+		foreach ( $results as $result ) {
+			$alloptions[ $result->option_name ] = maybe_unserialize( $result->option_value );
+		}
+
+		wp_cache_set( 'alloptions', $alloptions, 'options' );
+
+		$option_values = array();
+		foreach ( $results as $result ) {
+			$option_values[ $result->option_name ] = maybe_unserialize( $result->option_value );
+		}
+
+		wp_cache_set_multiple( $option_values, 'options' );
+	}
+}
+
+/**
+ * Wrapper function to prime all options registered with a specific option group.
+ *
+ * @since 6.4.0
+ *
+ * @global array $new_allowed_options
+ *
+ * @param string $option_group The option group to prime options for.
+ */
+function prime_options_by_group( $option_group ) {
+	global $new_allowed_options;
```

Comment:
The `$new_allowed_options` global is only referenced in core when registering or unregistering settings. I'm unsure if this is the appropriate global to check, or if we should be checking the `$allowed_options` global (if at all)?

Reply:
@joemcgill I don't think the `$allowed_options` global is right to use here. That global only exists within the WP Admin context, and only when loading the `/wp-admin/options.php` screen (which mostly isn't used as a screen, but to submit option updates to).

We need to use `$new_allowed_options` here as that is used by registering settings, which as such is precisely where an option is associated with an option group, as the primary use-case here is plugins which would need to call this function in order to use option groups anyway. Even any core setting that is registered (which most are) will work using this approach.